File: SDKs/apigee.net/marketcheck-cars/2.01/java/lib/src/main/java/openapisdk/models/shared/BasePopular.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * BasePopular
 * Single popular car in array
**/
public class BasePopular {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("city")
    public String city;
    public BasePopular withCity(String city) {
        this.city = city;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("counts")
    public String counts;
    public BasePopular withCounts(String counts) {
        this.counts = counts;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("dom_stats")
    public SalesStats domStats;
    public BasePopular withDomStats(SalesStats domStats) {
        this.domStats = domStats;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("inventoryType")
    public String inventoryType;
    public BasePopular withInventoryType(String inventoryType) {
        this.inventoryType = inventoryType;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("make")
    public String make;
    public BasePopular withMake(String make) {
        this.make = make;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("miles_stats")
    public SalesStats milesStats;
    public BasePopular withMilesStats(SalesStats milesStats) {
        this.milesStats = milesStats;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("model")
    public String model;
    public BasePopular withModel(String model) {
        this.model = model;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("price_stats")
    public SalesStats priceStats;
    public BasePopular withPriceStats(SalesStats priceStats) {
        this.priceStats = priceStats;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("state")
    public String state;
    public BasePopular withState(String state) {
        this.state = state;
        return this;
    }
}

File: SDKs/apigee.net/marketcheck-cars/2.01/python/src/sdk/models/shared/basepopular.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import salesstats as shared_salesstats


@dataclass_json
@dataclasses.dataclass
class BasePopular:
    r"""BasePopular
    Single popular car in array
    """
    
    city: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('city') }})
    counts: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('counts') }})
    dom_stats: Optional[shared_salesstats.SalesStats] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('dom_stats') }})
    inventory_type: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('inventoryType') }})
    make: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('make') }})
    miles_stats: Optional[shared_salesstats.SalesStats] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('miles_stats') }})
    model: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('model') }})
    price_stats: Optional[shared_salesstats.SalesStats] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('price_stats') }})
    state: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('state') }})
    
